stream-tests/src/test/java/org/apache/pekko/stream/javadsl/FlowTest.java

```diff
@@ -237,6 +238,22 @@ public void mustBeAbleToUseMapWithResource() {
     Assert.assertFalse(gate.get());
   }
 
+  @Test
+  public void mustBeAbleToUseMapWithAutoCloseableResource() {
+    final TestKit probe = new TestKit(system);
+    final AtomicInteger closed = new AtomicInteger();
+    Source.from(Arrays.asList("1", "2", "3"))
+        .via(
+            Flow.of(String.class)
+                .mapWithResource(
+                    () -> (AutoCloseable) closed::incrementAndGet, (resource, elem) -> elem))
+        .runWith(Sink.foreach(elem -> probe.getRef().tell(elem, ActorRef.noSender())), system);
```

Comment:
The StreamTestkit can be used in Java too, so we can avoid the `probe.getRef`, it's fine too, up to you:)

Reply:
Hmm actually I just copy&paste this `probe.getRef` from other test 😅

I don't know how I can avoid this `probe.getRef` so If you suggest some example code then I'll update it~! thanks!

Reply:
I knew it, because that me.